python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py

```diff
@@ -211,3 +214,92 @@ def _missing_value_repair(df):
         return False
     logging.warning("Missing data has be imputed.")
     return True
+
+
+def _abnormal_value_check(df, dt_col, id_col, threshold=10):
+    '''
+    This check is used to determine whether there are abnormal values in the data.
+    '''
+    for column in df.columns:
+        if column == dt_col or column == id_col:
+            continue
+        df_col = df[column]
+        flag = True
+        for val in df_col:
+            if isinstance(val, str):
+                flag = False
+                break
+        for val in df_col:
+            if flag is False:
+                break  # skip columns containing str
+            if df_col.std() != 0 and abs((val - df_col.mean()) / df_col.std()) > threshold:
+                logging.warning(f"Some values of column {column} exceeds the mean plus/minus "
+                                f"{threshold} times standard deviation, please call "
+                                f".repair_abnormal_data() to remove abnormal values.")
+                return False
```

Comment:
we have a really high complexity in this function.
For line 228 - 231, we have O(N)
For line 232 - 239, we have O(N^2) since for every iteration, we calculate std and mean.

I recommend to:
1. no need to have the first loop (started from 228), you may only use `df_col.dtype`
2. calculate std and mean before the second loop
3. try some vectorized method here rather than a for loop

Reply:
same comment to other implements